src/astro/sql/operators/save_file.py

```diff
@@ -0,0 +1,86 @@
+from typing import Optional, Union
+
+import pandas as pd
+from airflow.models import BaseOperator
+from airflow.models.xcom_arg import XComArg
+
+from astro.constants import ExportExistsStrategy
+from astro.databases import create_database
+from astro.files import File
+from astro.sql.table import Table
+from astro.utils.task_id_helper import get_task_id
+
+
+class SaveFile(BaseOperator):
+    """Write SQL table to csv/parquet on local/S3/GCS.
+
+    :param input_data: Table to convert to file
+    :param output_file: File object containing the path to the file and connection id.
+    :param if_exists: Overwrite file if exists. Default False.
+    """
+
+    template_fields = ("input_data", "output_file")
+
+    def __init__(
+        self,
+        input_data: Union[Table, pd.DataFrame],
+        output_file: File,
+        if_exists: ExportExistsStrategy = "exception",
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.output_file = output_file
+        self.input_data = input_data
```

Comment:
I am worried that our interfaces for `load` and `save` are not symmetric ATM:
* `load`: from file to table-only
* `save`: from table to either file or database

Normalizing this would be a breaking change to `load.`

@utkarsharma2 @kaxil @dimberman I'd love your thoughts on this:
1. Do we want to normalize?
2. If yes to (1), as part of this or in a separate PR?
3. If separate PR (2), when?

Reply:
I just spoke to @dimberman , he gave a good suggestion: if the user does not specify an `output_table` in `load_file`, we load to a dataframe.

@mikeshwe , feel free to loop in this discussion!

The suggestion is:
```
dataframe = aql.loadfile(File("s3://some-bucket/some-prefix"))
```

Reply:
I do like the sleek, minimalist look of this function! The current load_file requires an output_table parameter, AFAIK, which is unfortunate if we are just loading to a table object and don't care what the table is in the database, e.g,:

`orders_data = aql.load_file(
        path=S3_FILE_PATH + '/orders_data_header.csv',
        file_conn_id=S3_CONN_ID,
        output_table=Table(database=SNOWFLAKE_DATABASE,conn_id=SNOWFLAKE_CONN_ID))`

The part where I'm currently confused is that does it matter whether this:

`aql.loadfile(File("s3://some-bucket/some-prefix"))` 

returns a dataframe or a Table? Because even if we want to return a Table object, it seems unnecessary to have to specify the output_table parameter, because the SDK should just take care of that in the background--at most we should only have to globally specify some default backend for the SDK to do its magic, like in a .env variable.

IOW, it would be great if the same minimalist call to `load_file` could return either a dataframe or a Table object--maybe we have a boolean or enumerated type to specify which one?

Reply:
@mikeshwe thanks for your contribution! If we make the `output_table` argument optional and introduce the boolean argument, how do you suggest we retrieve the connection to create the temporary table ?

Reply:
@tatiana 
1. Do they need to be symmetric? Also, if we didn't have the need so far, why do we need it now?
2. I think there are easier alternatives to getting a data frame from a file right, pd.readCSV() etc. It offers little value.

Reply:
@utkarsharma2 I disagree. We are still handling the complexity and keeping stuff within our ecosystem. This will also offer value when we later offer lineage and data validation checks. Telling people "well just write a python function and use pandas.read_csv" would be asking them to leave our ecosystem and miss out on later benefits.

Reply:
Also if they later decide they want to move that data to a database they would only need to add an output_table vs. they would need to rewrite everything if they use pandas.read_csv. We are also handling the complexity around pulling from a variety of datastores.

Reply:
I like @dimberman 's suggestion of an optional `output_table` param and do agree with the argument of one interface instead of recommend pd.read_csv